src/pages/Send/StepOneForm.js

```diff
@@ -0,0 +1,388 @@
+        isGasLimitLoading: false,
+        isValidationFailed: true,
+      })
+      // #TODO: check network error, and show toast
+      // alert(`#TODO: Fixme when toast notifications will be ready\n
+      // Network error, can\'t request gasPrice`)
+
+      return String(staticGasAmount || FALLBACK_GAS_AMOUNT)
+    }
+  }
+
+  renderSendStepForm = (props: FormRenderProps) => {
+    const {
+      handleSubmit,
+      submitting: isSubmitting,
+      valid: isValid,
+      values,
+    } = props
+
+    const {
+      isGasPriceLoading,
+      isGasLimitLoading,
+    } = this.state
+
+    const {
+      recipientAddress,
+      assetAddress,
+      amountValue,
+      gasPriceValue,
+      gasLimitValue,
+    } = values || {}
+
+    const isPriorityEnabled = !!recipientAddress &&
+      !!assetAddress &&
+      !!amountValue
+
+    const isFormDisabled = !recipientAddress ||
+      !assetAddress ||
+      !amountValue ||
+      isGasLimitLoading ||
+      isGasPriceLoading ||
+      !isValid
+
+    const isEthereumAsset = checkETH(assetAddress)
+
+    return (
+      <form
+        onSubmit={handleSubmit}
+        className={styles.form}
+      >
+        <Field
+          className={styles.mb16}
+          component={ConnectedAssetPicker}
+          name='assetAddress'
+        />
+        <Field
+          className={styles.mb16}
+          component={ConnectedRecipientPicker}
+          name='recipientAddress'
+        />
+        <Field
+          className={styles.mb16}
+          component={ConnectedSendAmountField}
+          assetAddress={assetAddress}
+          gasPrice={gasPriceValue}
+          gasLimit={gasLimitValue}
+          name='amountValue'
+        />
+        <Field
+          isLoading={isGasPriceLoading}
+          isEth={isEthereumAsset}
+          isDisabled={!isPriorityEnabled}
+          className={styles.mb32}
+          component={PriorityField}
+          name='isPriorityOpen'
+          gasPriceFieldName='gasPriceValue'
+          gasLimitFieldName='gasLimitValue'
+        />
+        <Button
+          type='submit'
+          name='send'
+          isLoading={isSubmitting}
+          isDisabled={isFormDisabled}
+        >
+          {t`Next`}
+        </Button>
+      </form>
+    )
+  }
+
+  calculate = createCaluclateDecorator({
+    field: 'assetAddress',
+    updates: {
+      amountValue: () => '',
+      isPriorityOpen: () => false,
+      gasPriceValue: () => this.requestGasPrice(),
+      gasLimitValue: (_, allValues: ?SendFormValues) => allValues
+        ? this.requestGasLimit(allValues)
+        : String(FALLBACK_GAS_AMOUNT),
+    },
+  }, {
+    field: 'isPriorityOpen',
+    updates: {
+      gasPriceValue: () => this.requestGasPrice(),
+      gasLimitValue: (_, allValues: ?SendFormValues) => allValues
+        ? this.requestGasLimit(allValues)
+        : String(FALLBACK_GAS_AMOUNT),
+    },
+  }, {
+    field: 'recipientAddress',
+    updates: {
+      gasPriceValue: () => this.requestGasPrice(),
```

Comment:
please check that `this.requestGasPrice` is not called for each entered symbol in input

Reply:
It works correctly